src/libs/actions/IOU.ts

```diff
@@ -6600,6 +6601,56 @@ function deleteMoneyRequest(transactionID: string | undefined, reportAction: Ony
         value: null,
     });
 
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.SET,
+            key: `${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`,
+            value: transaction ?? null,
+        },
+    ];
+
+    if (transactionViolations) {
+        removeSettledAndApprovedTransactions(transactionViolations.find((violation) => violation?.name === CONST.VIOLATIONS.DUPLICATED_TRANSACTION)?.data?.duplicates ?? []).forEach(
```

Comment:
```suggestion
        const duplicateTransactionIDs = transactionViolations
            .filter((violation) => violation?.name === CONST.VIOLATIONS.DUPLICATED_TRANSACTION)
            .map((violation) => violation?.data?.duplicates ?? [])
            .flat();
        removeSettledAndApprovedTransactions(duplicateTransactionIDs).forEach((duplicateID) => {
```
Could you please update the logic to handle by getting `duplicateTransactionIDs` earlier?